### ISSUE TITLE
Project Roadmap / Status

### ISSUE THREAD

**Issue by heddendorp** [angular/angular-electron#23]
I just found this and it looks very cool to integrate electron deeper into angular.
I'd like to know if this project will be continued or if I should just use angular as is with electron?

**Comment by danielnieto** [comment]
:( I was very excited as well when I found it, it seems like it's dead, for me it doesn't even run.

**Comment by chadbr** [comment]
@robwormald Rob - any plans for this or?

**Comment by micaelgallego** [comment]
It is very sad that this project have no activity...

**Comment by elliotlarson** [comment]
@micaelgallego Agreed.  It's kind of a bummer that this project is stalled out, but even more of a bummer that the core team isn't really responding.  This is sort of the "official" agular2-electron library.  I get that the team is busy working on more important/pressing/interesting problems right now.  And I totally applaud and appreciate their efforts I'm seeing in other repos.  But it would be nice to just get an update or reply here.

**Comment by chadbr** [comment]
the only alternative I'm aware of is https://github.com/electron-userland/electron-forge

**Comment by robwormald** [comment]
So an update here:

I've just started work on rewriting this plugin to work with 2.4.x+ of Angular, with the *eventual* goal of making this a first-party platform. I've pushed a `development` branch to this repo, but please do *not* report bugs or issues against it yet. I don't have an ETA on when this will be ready yet.

We (Angular team) have a finite numbers of hours in the day, and this project is, for the moment, not at the top of the priority list. If you're well versed in Electron or Angular's rendering architecture, I'd love to have you contribute here. 

From a roadmap POV, this is something I'd like to hear from the community on. There are a few different models that I can see here:
- currently this repo implements a split rendering setup, like WebWorkers, where we run Angular in the main (node) process and use IPC to communicate with renderer (Browser) windows. 
- I'd like to, ASAP, support the multi-window architecture. It's very doable but non-trivial.
- others may want to use electron in the more "traditional" sense where Angular runs fully in the Browser and simply uses IPC to talk to node. The best help we can offer here is a suite of Angular friendly Electron services and proper ZoneJS integration.
- some combination of the above
- "headless" cases (task tray apps and such)
- insert your use case here

Long term, I'd like to see this repo go through the same sort of process as Universal - some time as a development proof-of-concept eventually graduating to full core team support.

**Comment by chadbr** [comment]
Thanks for the update Rob. FWIW, at the moment - I'm looking for option 3 & 2 (in that order).

All the other scenarios are great options...

**Comment by leifwells** [comment]
@robwormald  

I have some requests:

- Can you video/present what this project represents? With Angular's maturity, I feel that key concepts have also matured as have the communication of those key concepts. I, personally, have lost the thread as to where Electron's Renderer Process ends and your proposed Angular Renderer begins. Perhaps this might be a good way to begin the reboot of the project, set focus, establish goals, generate buzz and rally developers.
- Can you document your thoughts on what it might take to get multi-window support working? Should we start with single window and move to multi-window? Parallel development? 
- Can we make sure that it is possible to use angular-electron with and without the [Angular CLI](https://github.com/angular/angular-cli)? Personally, I enjoy using the CLI, but I'm sure there are folks out there who may not. 
- Can we get the [Electron Forge CLI](https://github.com/electron-userland/electron-forge) integrated as well? I really like that CLI, as it simplifies install, development and building headaches.
- Can we establish some Best Practices when it comes to building an angular-electron application? As in, where to put our Angular source code, where to place compiled code for Electron to use in development, how to insure a production build will generate all the artifacts required for an Electron application? A lot of this guidance could probably happen during development of a sample application. _Side note: As an example, I'm currently having a difficult time getting embedded fonts to load and display properly._

I am currently working on a small sample project that could also evolve into something that would easily be applicable to angular-electron. While I am not "well versed in Electron or Angular's rendering architecture", I am willing to assist in any way. 

Thanks for the update.

**Comment by seiyria** [comment]
Definitely, I think the multi-window aspect is a great idea and I would use this if so. I built a point of sale system and I had to do some very hokey things. Basically, in a browser testing environment, `window.open` worked really well, but when I wanted to test in electron, well, [I had to do this](https://github.com/seiyria/posys/blob/master/src/client/pages/reporting/reporting.ts#L238-L283). It's definitely not something I'm proud of, but it works.

What's worse, is that because of how non-trivial it is, I settled for making the report popup incredibly dumb, so I [generate the HTML beforehand](https://github.com/seiyria/posys/blob/master/src/client/pages/reporting/reportdatatransformer.ts#L24-L56) and have some styles to make it not look terrible. I wasn't feeling up to the challenge of getting angular or bootstrap in there, but if I had them, it would have been a lot cleaner.

**Comment by steven-chou** [comment]
I just started an Electron app with the latest Angular 4. I have many use cases that the new BrowserWindow needs to be created/opened though the Electron main process (I think this is what the Electron documentation suggests) when the user clicks the button or select a menu item. Since I can't even launch this demo application locally, I can't really tell if opening a new BrowserWindow inside the Angular is supported. (I guess from @robwormald said earlier, this seems to be on the future road map) 

Given the current situation, what will be my best bet to handle this kind of use case in Angular? 
Does making every window a separate Angular app a viable solution? (That sounds a crazy solution myself)

**Comment by heddendorp** [comment]
I'd like to follow up on this issue, is there new information on your side @robwormald ?
It sounds to me like the webworker approach could provide a great platform especially when used with ngrx, although there seem to be problems with the current webworker mode as well.

If this project is still low priority you should consider removing it from https://angular.io/resources until it's updated.
I really hope this is continued soon ;)

**Comment by Kaffiend** [comment]
With the recent patch to electron for zone.js, can we expect anything on this front in the near future?
